Sources/Sentry/SentryPropagationContext.h

```diff
@@ -1,19 +1,29 @@
 #import <Foundation/Foundation.h>
 
-@class SentryId, SentrySpanId, SentryTraceContext, SentryTraceHeader;
+@class SentryId, SentrySpanId, SentryTraceContext, SentryTraceHeader, SentryOptions, SentryScope;
 
 NS_ASSUME_NONNULL_BEGIN
 
 @interface SentryPropagationContext : NSObject
 
 @property (nonatomic, strong) SentryId *traceId;
 @property (nonatomic, strong) SentrySpanId *spanId;
-@property (nonatomic, readonly, nullable) SentryTraceContext *traceContext;
-
 @property (nonatomic, readonly) SentryTraceHeader *traceHeader;
 
 - (NSDictionary<NSString *, NSString *> *)traceContextForEvent;
 
+/**
+ * We need to pass in the options and the user segment instead of retrieving them from the static
+ * API cause, for example, the SentryClient can have multiple instances with different options and
+ * scopes running and in that scenario we need can't retrieve these values from the static API.
+ *
+ *  @param options The current active options.
+ *  @param userSegment You can retrieve this usually from the `scope.userObject.segment`.
+ */
+- (SentryTraceContext *)getTraceContext:(SentryOptions *)options
```

Comment:
The way this is named makes me think it's retrieving something like from an ivar, but really it's creating a new thing every time it's called. Can we rename it to e.g. 'traceContextForOptions:userSegment:`?

Reply:
Really, I would prefer that we just call `[[SentryTraceContext alloc] initWithTraceId:...]` in the places where this method is called, I don't understand why this should be wrapped in `SentryPropagationContext` at all, the indirection at the call sites is confusing.

Reply:
That is an excellent point. I fully I agree 🥇. I moved it.